notebooks/ir_ima_visualization/IR_IMA_Visualization_with_an_Example_of_Time_Variable_Background.ipynb

```diff
@@ -0,0 +1,923 @@
+    "    rhs_region={\"x0\":700,\"x1\":900,\"y0\":100,\"y1\":900}\n",
+    "    m,l,r=diff.getmedlhsrhs(cube, lhs_region, rhs_region)\n",
+    "    md, ld, rd, ms, ls, rs = diff.cumulative_diff(cube, integ_time, lhs_region, rhs_region)\n",
+    "\n",
+    "    fig_0=panel.panel_plot(cube, integ_time,md,ld,rd,ms,ls,rs, diff_method='cumulative')\n",
+    "    _=fig_0.suptitle(filename, fontsize=40)\n",
+    "    plt.subplots_adjust(bottom=0.25, right=0.9, top=0.95)\n",
+    "\n",
+    "except NameError:\n",
+    "    print(\"No file by this name found\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we expect, the differences images are blank (with the exception of a speckled pattern showing cosmic rays and differences in noise between reads). Both 'delta'($\\Delta$) and $\\mu$ are effectively zero in this nominal image. The ratio, in this case, is less useful becuase the signal differences between left and right are close to zero. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. A More Accurate Difference Between Reads: The Instantaneous Rate <a id=\"instantaneous\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we implement a different technique to examine the count rate difference between consecutive reads. In this case, we first convert from count rate (e-/s) back to counts (e-) before taking the difference, as shown in equation 3 from [WFC3/IR ISR 2018-05](https://www.stsci.edu/files/live/sites/www/files/home/hst/instrumentation/wfc3/documentation/instrument-science-reports-isrs/_documents/2018/WFC3-2018-05.pdf).\n",
+    "\n",
+    "![Instantaneous Difference Equation](instantaneous_diff.png)\n",
+    "\n",
+    "One advantage of this method is that it does not remove information about the signal levels at the beginning of the exposure, as in the prior method. Thus the ramp plot below shows a mean rate of ~1.5 e-/s for the first two differences instead of ~zero. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = plt.figure(figsize=(50, 20))\n",
+    "fig\n",
+    "rows = 1\n",
+    "columns = 2\n",
+    "\n",
+    "ima_files=[ima_scattered, ima_nominal] #If only using one image, please remove the extraneous image from this list \n",
+    "\n",
+    "for i,ima in enumerate(ima_files):\n",
+    "    \n",
+    "    head, tail = os.path.split(ima)\n",
+    "    filename=tail\n",
+    "    cube, integ_time =diff.read_wfc3(ima)\n",
+    "\n",
+    "    #Below, we are implementing upper and lower limits on our pixel values to exclude sources when plotting our ramp.\n",
+    "    #Please use a limit that makes sense for your own data, when running your images through this notebook.\n",
+    "\n",
+    "    with np.nditer(cube, op_flags=['readwrite']) as it:\n",
```

Comment:
If you're changing all values above/below 3/-3, this might be a more condense way to do that:
```
cube[np.abs(cube) > 3] = np.nan
```
Also how is x being called?